packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts

```diff
@@ -0,0 +1,151 @@
+        {
+          principalId: `${PRINCIPAL_ID_EXPECTED}`,
+          clientId: `${CLIENT_ID_EXPECTED}`,
+        },
+    },
+    principalId: `${PRINCIPAL_ID_EXPECTED}`,
+    tenantId: `${TENANT_ID_EXPECTED}`,
+  },
+  properties: {
+    provisioningState: "Succeeded",
+  },
+};
+
+// managed identity tracked resource
+Scenarios.Azure_ResourceManager_Models_CommonTypes_ManagedIdentity_ManagedIdentityTrackedResources_get = passOnSuccess([
+  mockapi.get(
+    "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Models.CommonTypes.ManagedIdentity/managedIdentityTrackedResources/:managedIdentityResourceName",
+    (req) => {
+      req.expect.containsQueryParam("api-version", "2023-12-01-preview");
+      if (req.params.subscriptionId !== SUBSCRIPTION_ID_EXPECTED) {
+        throw new ValidationError("Unexpected subscriptionId", SUBSCRIPTION_ID_EXPECTED, req.params.subscriptionId);
+      }
+      if (req.params.resourceGroup.toLowerCase() !== RESOURCE_GROUP_EXPECTED) {
+        throw new ValidationError("Unexpected resourceGroup", RESOURCE_GROUP_EXPECTED, req.params.resourceGroup);
+      }
+      if (req.params.managedIdentityResourceName.toLowerCase() !== "identity") {
+        throw new ValidationError(
+          "Unexpected managed identity resource name",
+          "identity",
+          req.params.managedIdentityResourceName,
+        );
+      }
+      return {
+        status: 200,
+        body: json(validSystemAssignedManagedIdentityResource),
+      };
+    },
+  ),
+]);
+
+Scenarios.Azure_ResourceManager_Models_CommonTypes_ManagedIdentity_ManagedIdentityTrackedResources_createWithSystemAssigned =
+  passOnSuccess([
+    mockapi.put(
+      "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Models.CommonTypes.ManagedIdentity/managedIdentityTrackedResources/:managedIdentityResourceName",
+      (req) => {
+        req.expect.containsQueryParam("api-version", "2023-12-01-preview");
+        if (req.params.subscriptionId !== SUBSCRIPTION_ID_EXPECTED) {
+          throw new ValidationError("Unexpected subscriptionId", SUBSCRIPTION_ID_EXPECTED, req.params.subscriptionId);
+        }
+        if (req.params.resourceGroup.toLowerCase() !== RESOURCE_GROUP_EXPECTED) {
+          throw new ValidationError("Unexpected resourceGroup", RESOURCE_GROUP_EXPECTED, req.params.resourceGroup);
+        }
+        if (req.params.managedIdentityResourceName.toLowerCase() !== "identity") {
+          throw new ValidationError(
+            "Unexpected managed identity resource name",
+            "identity",
+            req.params.managedIdentityResourceName,
+          );
+        }
+        req.expect.bodyEquals({
+          location: `${LOCATION_REGION_EXPECTED}`,
+          tags: {
+            tagKey1: "tagValue1",
+          },
+          properties: {},
```

Comment:
How's language SDK behave, when the `properties` property only contains read-only property?

Do it send the empty `properties` object, or not send `properties` at all?

Reply:
It send the empty properties object when the properties property only contains read-only property.

Reply:
Yeah, I am checking for language other than Java.

If some other language does it differently, we may better to add a writable property within.

Reply:
@v-hongli1  Or are you able to skip the check about `properties`?

Basically, do not do a full equal on body. Just check e.g. the `identities` property in the request body. This is what your test cases, not the other part of the request.

Reply:
@weidongxu-microsoft 
I have tried your idea, but found that the subjects must be full equality on the body to avoid causing a validation exception for the request data..

Reply:
@v-hongli1 I don't quite understand. Could you paste the code?

Reply:
@XiaofeiCao 
If it is a test case code, please refer to the following as:
```
    public void testManagedIdentityManager() {
        Map<String, String> tagsMap = new HashMap<>();
        tagsMap.put("tagKey1", "tagValue1");

        Map<String, UserAssignedIdentity> userAssignedIdentityMap = new HashMap<>();
        userAssignedIdentityMap.put(
                "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1",
                new UserAssignedIdentity()
        );

        ManagedIdentityTrackedResource resource = manager.managedIdentityTrackedResources()
                .define("identity")
                .withRegion(Region.US_EAST)
                .withExistingResourceGroup("test-rg")
                .withProperties(new ManagedIdentityTrackedResourceProperties())
                .withIdentity(new ManagedServiceIdentity().withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED))
                .withTags(tagsMap)
                .create();
        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED, resource.identity().type());

        resource = manager.managedIdentityTrackedResources().getById(resource.id());
        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED, resource.identity().type());

        resource.update().withIdentity(
                new ManagedServiceIdentity()
                        .withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED)
                        .withUserAssignedIdentities(userAssignedIdentityMap))
                .apply();
        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED, resource.identity().type());
        Assertions.assertNotNull(resource.identity().userAssignedIdentities());
    }
```

Reply:
I mean the code in `mockapi.ts`. You mentioned that you tried not to do a full equal on body, but still can't get the verification pass?

Reply:
You may not able to use the ready method like `req.expect.bodyEquals`

Just check the source of the func, likely do something like
```
  if (!deepEqual(request.body["identities"], expectedIdentity, { strict: true })) {
    throw new ValidationError("message", expectedIdentity, request.body["identities"]);
  }
```

Reply:
> You may not able to use the ready method like `req.expect.bodyEquals`
> 
> Just check the source of the func, likely do something like
> 
> ```
>   if (!deepEqual(request.body["identities"], expectedIdentity, { strict: true })) {
>     throw new ValidationError("message", expectedIdentity, request.body["identities"]);
>   }
> ```

Fixed following as : https://github.com/Azure/cadl-ranch/pull/609/commits/0e090c459b96f2162eac2bacb5ec89754f924b3a